src/utils/hooks/useGetData.js

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useGetData(URL) {
+    const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
+    const [getData, setGetData] = useState(() => ({
+        data: {},
+        isLoading: true,
+    }));
+
+    useEffect(() => {
+        if (!apiRef || isApiMetadataLoading) {
+            return () => { };
+        }
+
+        const controller = new AbortController();
+
+        async function getGetData() {
+            try {
+                setGetData({ data: {}, isLoading: true });
+
+                const response = await fetch(
+                    URL,
+                    {
+                        signal: controller.signal,
+                    }
+                );
+                const data = await response.json();
+
+                setGetData({ data, isLoading: false });
+            } catch (err) {
+                setGetData({ data: {}, isLoading: false });
+                console.error(err);
+            }
+        }
+
+        getGetData();
+
+        return () => {
+            controller.abort();
+        };
+    }, [apiRef, isApiMetadataLoading, URL]);
+
+    return getData;
```

Comment:
You could also return the loading flag, something like: `return {isloading, getData}` or `return [isLoading, getData]` so you can have a way to tell the user something is loading backside for example with a loader.

Reply:
Haha nevermind, I just saw you did so :)